google-datacatalog-connectors-commons/src/google/datacatalog_connectors/commons/datacatalog_facade.py

```diff
@@ -126,20 +129,83 @@ def __entry_was_updated(cls, current_entry, new_entry):
+        object_2.user_specified_type = new_entry.user_specified_type
+        object_2.display_name = new_entry.display_name
+        object_2.description = new_entry.description
+        object_2.linked_resource = new_entry.linked_resource
+
+        return object_1 == object_2 and cls.__schemas_are_equal(
+            current_entry.schema, new_entry.schema)
+
+    @classmethod
+    def __schemas_are_equal(cls, current_schema, new_schema):
+        current_columns = current_schema.columns
+        new_columns = new_schema.columns
+
+        columns_are_equal = set(
+            [current_column.column for current_column in current_columns]) == \
+            set(new_column.column for new_column in new_columns)
+
+        # If columns don't match return early for optimization
+        # for example in case a column is deleted or
+        # a new column is created.
+        if not columns_are_equal:
+            return False
+
+        for new_column in new_columns:
+            found_column = next(
+                (current_column for current_column in current_columns
+                 if current_column.column == new_column.column), None)
+
+            if not found_column:
+                return False
+
+            if not cls.__columns_fields_are_equal(found_column, new_column):
+                return False
+
+        return True
+
+    @classmethod
+    def __columns_fields_are_equal(cls, current_column, new_column):
         object_1 = utils.ValuesComparableObject()
-        object_1.user_specified_system = entry_1.user_specified_system
-        object_1.user_specified_type = entry_1.user_specified_type
-        object_1.display_name = entry_1.display_name
-        object_1.description = entry_1.description
-        object_1.linked_resource = entry_1.linked_resource
+        object_1.column = current_column.column
+        object_1.description = current_column.description
+        object_1.type = current_column.type
+
+        # If it is not fulfilled we need to initialize with the default MODE.
+        if not current_column.mode:
+            current_column.mode = cls.__DEFAULT_COLUMN_MODE
+
+        object_1.mode = current_column.mode
+
+        # Currently we only look at the subcolumns length
+        # connectors do not utilize this field at the moment.
+        object_1.subcolumns_len = len(current_column.subcolumns)
 
         object_2 = utils.ValuesComparableObject()
-        object_2.user_specified_system = entry_2.user_specified_system
-        object_2.user_specified_type = entry_2.user_specified_type
-        object_2.display_name = entry_2.display_name
-        object_2.description = entry_2.description
-        object_2.linked_resource = entry_2.linked_resource
+        object_2.column = new_column.column
+        object_2.description = new_column.description
+        object_2.type = new_column.type
+
+        # If it is not fulfilled we need to initialize with the default MODE.
```

Comment:
```suggestion
        # We need to initialize with the default MODE if it is not fulfilled.
```
Nitpick... makes reading more friendly. 😃